charts/partials/templates/_network-validator.tpl

```diff
@@ -3,8 +3,11 @@ name: linkerd-network-validator
 image: {{.Values.proxy.image.name}}:{{.Values.proxy.image.version | default .Values.linkerdVersion }}
 imagePullPolicy: {{.Values.proxy.image.pullPolicy | default .Values.imagePullPolicy}}
 {{ include "partials.resources" .Values.proxy.resources }}
-{{- if or .Values.networkValidator.enableSecurityContext }}
+{{- if ne .Values.networkValidator.enableSecurityContext false }}
```

Comment:
is this the same as `if .Values.networkValidator.enableSecurityContext` ?

Reply:
apparently not. We are removing this value from values.yaml. if we do if `.Values.networkValidator.enableSecurityContext`, this will always triger in the default case because this value is not set. What we are doing here is that we are testing explicitly for the case when someone sets this value to false. We want to do that in order to preserve backwards compatibility. Does that make sense? 

FWIW I was surprised by this behavior...